lib/OpenQA/Worker/Job.pm

```diff
@@ -0,0 +1,1263 @@
+
+        my $current_test_module         = $self->current_test_module;
+        my $is_last_test_to_be_uploaded = $remaining_test_count eq 1 || $test eq $upload_up_to;
+        my $test_not_running            = !$current_test_module || $test ne $current_test_module;
+        my $test_is_completed           = !$is_last_test_to_be_uploaded || $test_not_running;
+        if ($test_is_completed) {
+            # remove completed tests from @$test_order so we don't upload those results twice
+            shift(@$test_order);
+        }
+
+        last unless ($result);
+        $ret{$test} = $result;
+
+        if ($result->{extra_test_results}) {
+            for my $extra_test (@{$result->{extra_test_results}}) {
+                my $extra_result = read_module_result($extra_test->{name});
+                next unless $extra_result;
+                $ret{$extra_test->{name}} = $extra_result;
+            }
+            push @{$extra_test_order}, @{$result->{extra_test_results}};
+        }
+
+        last if $is_last_test_to_be_uploaded;
+    }
+    return \%ret;
+}
+
+sub _read_module_result {
+    my ($self, $test) = @_;
+
+    my $result = $self->_read_json_file("result-$test.json");
+    return unless $result;
+
+    my $images_to_send = $self->images_to_send;
+    my $files_to_send  = $self->files_to_send;
+    for my $d (@{$result->{details}}) {
+        if ($d->{json}) {
+            $d->{json} = $d->{json};
+        }
+        for my $n (@{$d->{needles}}) {
+            $n->{json} = $n->{json};
+
+        }
+        for my $type (qw(screenshot audio text)) {
+            my $file = $d->{$type};
+            next unless $file;
+
+            if ($type eq 'screenshot') {
+                my $md5 = $self->_calculate_file_md5("testresults/$file");
+                $d->{$type} = {
+                    name => $file,
+                    md5  => $md5,
+                };
+                $images_to_send->{$md5} = $file;
+            }
+            else {
+                push(@$files_to_send, $file);
+            }
+        }
+    }
+    return $result;
+}
+
+sub _calculate_file_md5 {
+    my ($self, $file) = @_;
+
+    my $c   = path($self->worker->pool_directory, $file)->slurp;
+    my $md5 = Digest::MD5->new;
+    $md5->add($c);
+    return $md5->clone->hexdigest;
+}
+
+sub _read_last_screen {
+    my ($self) = @_;
+
+    my $pooldir  = $self->worker->pool_directory;
+    my $lastlink = readlink("$pooldir/qemuscreenshot/last.png");
+    return if !$lastlink || $self->last_screenshot eq $lastlink;
+    my $png = encode_base64(path($pooldir, "qemuscreenshot/$lastlink")->slurp);
+    $self->{_last_screenshot} = $lastlink;
+    return {name => $lastlink, png => $png};
+}
+
+sub _log_snippet {
+    my ($self, $file, $offset_name) = @_;
+
+    my $offset = $self->$offset_name;
+    my $fd;
+    my %ret;
+    unless (open($fd, '<:raw', $file)) {
+        return \%ret;
+    }
+
+    sysseek($fd, $offset, Fcntl::SEEK_SET);    # FIXME: handle error?
```

Comment:
Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit or ticket